caikit_nlp/modules/text_generation/text_generation_local.py

```diff
@@ -562,41 +571,43 @@ def _preprocess_function(
         max_source_length: int,
         max_target_length: int,
         shuffle: bool,
+        use_iterable_dataset: bool,
     ):
         """Pre-process each example to get it prepared for training."""
-
-        # TODO: We are using a default verbalizer which is strictly tied to
-        # source training record currently. We need to figure out a better
-        # way to make verbalizer optional for build_task_tokenize_function
-        (
-            tokenize_function,
-            requires_unwrapping,
-        ) = base_model.build_task_tokenize_function(
-            tokenizer, max_source_length, max_target_length, verbalizer="{{input}}"
-        )
-
-        # dataset = datasets.load_dataset("billsum", split="ca_test", streaming=True)
-        # train_test_dataset = dataset.train_test_split(test_size=0.2)
-        dataset = TransformersIterableDataset.from_generator(
-            get, gen_kwargs={"train_stream": train_stream}
-        )
-
+        if use_iterable_dataset:
+            # Generator based
+            log.debug("Loading data as an iterable dataset")
+            dataset = TransformersIterableDataset.from_generator(
```

Comment:
Sure, that sounds good to me!

We might want to think about what this means for causal LM in the future. It's kind of weird for the way we now handle tokenization, because 1 step really means 1 example, but currently only example produces multiple things to learn on. I think

Reply:
hmm.. 1 step would mean 1 batch I think 🤔

Reply:
Whoops! You're right - ignore me :)